backend/service/topology/cache.go

```diff
@@ -0,0 +1,49 @@
+package topology
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// Perfroms leader election by acquiring a postgres advisory lock.
+// Once the lock is acquired topology caching is started.
+func (c *client) acquireTopologyCacheLock() {
```

Comment:
Still trying to craft some test for this, might refactor a bit but overall its ready for initial review.